gateway/api_loader.go

```diff
@@ -918,12 +937,15 @@ func (gw *Gateway) loadApps(specs []*APISpec) {
 
 	gw.DefaultProxyMux.swap(muxer, gw)
 
-	// Swap in the new register
 	gw.apisMu.Lock()
 
-	// release current specs resources before overwriting map
-	for _, curSpec := range gw.apisByID {
-		curSpec.Release()
+	for _, spec := range specs {
+		curSpec, ok := gw.apisByID[spec.APIID]
+		if ok && curSpec.Checksum != spec.Checksum {
+			curSpec.Release()
+		} else if ok && curSpec.Checksum == spec.Checksum {
+			spec.Release()
```

Comment:
spec is new spec - but it has already some resources allocated during MakeSpec phase even if it's not in use yet - so it's necessary to call spec.Release()

Reply:
Can you add it as a commend to the code? Otherwise it looks confusing.